pandas/core/base.py

```diff
@@ -684,8 +684,14 @@ def _gotitem(self, key, ndim, subset=None):
         # with the same groupby
         kwargs = dict([(attr, getattr(self, attr))
                        for attr in self._attributes])
+
+        try:
+            groupby = self._groupby[key]  # get slice of frame
+        except Exception:
```

Comment:
Ideally would catch something more descriptive than an `Exception`, but that is all that gets raised:

https://github.com/pandas-dev/pandas/blob/67e6e6fcd19d1d89cb60abc3a78372bc85fd8e29/pandas/core/base.py#L249

Maybe an AttributeError would be more appropriate?

Reply:
yeah see if u can catch something more specific bere